src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForArrayTests.cs

```diff
@@ -0,0 +1,1136 @@
+                {
+                    var c = (int[])[1];
+                }
+            }
+            """,
+            LanguageVersion = LanguageVersionExtensions.CSharpNext,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestNotTargetTypedWithIdentifierCast()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                using IntArray = int[];
+
+                class C
+                {
+                    void M()
+                    {
+                        var c = (IntArray)new int[] { 1 };
+                    }
+                }
+                """,
+            LanguageVersion = LanguageVersionExtensions.CSharpNext,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestTargetTypedInConditional1()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                class C
+                {
+                    void M(int[] x)
+                    {
+                        var c = true ? [|[|new|] int[]|] { 1 } : x;
+                    }
+                }
+                """,
+            FixedCode = """
+                class C
+                {
+                    void M(int[] x)
+                    {
+                        var c = true ? [1] : x;
+                    }
+                }
+                """,
+            LanguageVersion = LanguageVersionExtensions.CSharpNext,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestTargetTypedInConditional2()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                class C
+                {
+                    void M(int[] x)
+                    {
+                        var c = true ? x : [|[|new|] int[]|] { 1 };
+                    }
+                }
+                """,
+            FixedCode = """
+                class C
+                {
+                    void M(int[] x)
+                    {
+                        var c = true ? x : [1];
+                    }
+                }
+                """,
+            LanguageVersion = LanguageVersionExtensions.CSharpNext,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestTargetTypedInConditional3()
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                class C
+                {
+                    void M(int[] x)
+                    {
+                        int[] c = true ? null : [|[|new|] int[]|] { 1 };
+                    }
+                }
+                """,
+            FixedCode = """
+                class C
+                {
+                    void M(int[] x)
+                    {
+                        int[] c = true ? null : [1];
+                    }
+                }
+                """,
+            LanguageVersion = LanguageVersionExtensions.CSharpNext,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task TestTargetTypedInConditional4()
```

Comment:
```suggestion
    public async Task TestTargetTypedMissingInConditional4()
```

Reply:
sure.  can update that name in followup :)